src/Frontend/Page/Topic.hs

```diff
@@ -158,44 +146,64 @@ instance FormPage CreateTopic where
-        <*> pure ideas
+        <*> selectedIdeaParameters ideas
 
-    formPage v fa p =
+    formPage v fa p@(CreateTopic _space ideas) =
         semanticDiv p $ do
             h3_ "Create Topic"
             DF.form v fa $ do
                 DF.inputText     "title" v >> br_ []
                 DF.inputTextArea Nothing Nothing "desc" v >> br_ []
+                -- FIXME: There is no image in the click dummy.
                 DF.inputText     "image" v >> br_ []
+                ideaCheckboxes v ideas
                 DF.inputSubmit   "Add Topic"
 
-instance FormPage MoveIdeasToTopic where
+-- Edit topic description and add ideas to topic.
+data TopicFormPayload = TopicFormPayload ST Document [AUID Idea]
+  deriving (Eq, Show)
+
+
+instance FormPage EditTopic where
     -- While the input page contains all the wild ideas the result page only contains
     -- the ideas to be added to the topic.
-    type FormPageResult MoveIdeasToTopic = [AUID Idea]
+    type FormPageResult EditTopic = TopicFormPayload
 
-    formAction (MoveIdeasToTopic space topicId _) = relPath . U.Space space $ U.ListTopicIdeas topicId
-    redirectOf (MoveIdeasToTopic space topicId _) = relPath . U.Space space $ U.ListTopicIdeas topicId
+    formAction (EditTopic space topic _) = relPath . U.Space space $ U.MoveIdeasToTopic (topic ^. _Id)
+    redirectOf (EditTopic space topic _) = relPath . U.Space space $ U.ListTopicIdeas (topic ^. _Id)
 
-    makeForm (MoveIdeasToTopic _ _ ideas) =
-        fmap catMaybes . sequenceA $
-        [ justIf (idea ^. _Id) <$> (ideaToFormField idea .: DF.bool Nothing) | idea <- ideas ]
+    makeForm (EditTopic _space topic ideas) =
+        TopicFormPayload
+        <$> ("title" .: DF.text (Just (topic ^. topicTitle)))
+        <*> ("desc"  .: (Markdown <$> DF.text (Just $ fromMarkdown (topic ^. topicDesc))))
+        <*> selectedIdeaParameters ideas
 
-    formPage v fa p@(MoveIdeasToTopic _ _ ideas) = do
+    formPage v fa p@(EditTopic _space _topic ideas) = do
         semanticDiv p $ do
             h3_ "Wählen Sie weitere Ideen aus"
             DF.form v fa $ do
-                ul_ $ do
-                    for_ ideas $ \idea ->
-                        li_ $ do
-                            DF.inputCheckbox (ideaToFormField idea) v
-                            idea ^. ideaTitle . html
+                DF.inputText     "title" v >> br_ []
+                DF.inputTextArea Nothing Nothing "desc" v >> br_ []
+                ideaCheckboxes v ideas
                 DF.inputSubmit "Speichern"
                 button_ "Abbrechen" -- FIXME
 
 ideaToFormField :: Idea -> ST
 ideaToFormField idea = "idea-" <> cs (show $ idea ^. _Id)
 
+ideaCheckboxes :: (Monad m) => View (HtmlT m ()) -> [Idea] -> HtmlT m ()
+ideaCheckboxes v ideas =
+    ul_ . for_ ideas $ \idea ->
+        li_ $ do
+            DF.inputCheckbox (ideaToFormField idea) v
+            idea ^. ideaTitle . html
+
+selectedIdeaParameters :: forall m v . (Monad m, Monoid v)
```

Comment:
I think this may be a good application for nested forms.  But we don't have to investigate this idea here, just saying.